components/SearchRestaurantsForm/index.js

```diff
@@ -0,0 +1,40 @@
+import { StyledButton, StyledForm, StyledLegend } from "../styles";
+
+export default function SearchRestaurantsForm({
+  onMatchRestaurants,
+  restaurants,
+}) {
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const dataForRestaurantsSearch = Object.fromEntries(formData);
+    onMatchRestaurants(dataForRestaurantsSearch, restaurants);
+  }
+  return (
+    <StyledForm
+      aria-labelledby="formHeader"
+      onSubmit={(event) => handleSubmit(event)}
+    >
+      <StyledLegend id="formHeader">Suchen verfügbare Plätze</StyledLegend>
```

Comment:
A HTMl `legend` element[ is used to caption a `fieldset`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/legend), not the entire form. Please use a heading element instead.

Besides, the use of the `aria-` attributes is great!

Reply:
ah... I will update it